go/indexer/services/l1/confirmed_headers.go

```diff
@@ -128,47 +128,57 @@ type ConfirmedHeaderSelector struct {
 	cfg HeaderSelectorConfig
 }
 
-func toBlockNumArg(number *big.Int) string {
-	if number == nil {
-		return "latest"
+func HeadersByRange(ctx context.Context, client *rpc.Client, startHeight uint64, count int) ([]*NewHeader, error) {
+	height := startHeight
+	batchElems := make([]rpc.BatchElem, count)
+	for i := 0; i < count; i++ {
+		batchElems[i] = rpc.BatchElem{
+			Method: "eth_getBlockByNumber",
```

Comment:
Can save some bandwidth by using `eth_getHeaderByNumber` if you only want the header, see
https://github.com/ethereum/go-ethereum/blob/9fd8825d5a196edde6d8ef81382979875145b346/internal/ethapi/api.go#L716

Reply:
I'll refactor this separately.